File: C#/1786 - CPF 2.cs
/// author Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
/// copyright Josélio Júnior (Lunatic Fox) 2022

using System;
using System.Linq;
using System.Collections.Generic;
using System.Text.RegularExpressions;

class URI {
  static void Main(string[] args) { 
    while (true) {
      var r = Console.ReadLine() ?? "";
      if (r == "") break;
  
      string f9 = string.Join(".",
        Regex.Split(r, @"(\d{3})")
        .Where(e => !string.IsNullOrEmpty(e))
      );
  
      int[] r9 = r.ToCharArray()
        .Select(e => Int32.Parse(Convert.ToString(e)))
        .ToArray();
    
      List<int> _b1 = new List<int>(), _b2 = new List<int>();
      
      for (int i = 0; i < r9.Length; i++) {
        _b1.Add((i + 1) * r9[i]);
        _b2.Add((9 - i) * r9[i]);
      }
  
      int b1 = _b1.Aggregate((a, b) => a + b);
      int b2 = _b2.Aggregate((a, b) => a + b);
  
      b1 = b1 % 11 == 10 ? 0 : b1 % 11;
      b2 = b2 % 11 == 10 ? 0 : b2 % 11;
  
      Console.WriteLine($"{f9}-{b1}{b2}");
    }
  }
}

File: Pascal/1786 - CPF 2.pas
// author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
// copyright: Josélio Júnior (Lunatic Fox) 2023

var
  r: string;
  na: array[1..9] of integer;
  b1, b2, i, j, tna: integer;
begin
  while not Eoln do
  begin
    readln(r);
    b1 := 0;
    b2 := 0;
    j := 9;
    for i := 1 to 9 do
    begin
      tna := ord(r[i]) - 48;
      if (tna >= 0) and (tna <= 9) then
      begin
        b1 := b1 + (tna * i);
        b2 := b2 + (tna * j);
      end;
      j := j - 1;
    end;

    b1 := b1 mod 11;
    b2 := b2 mod 11;
    
    if (b1 = 10) then b1 := 0;
    if (b2 = 10) then b2 := 0;

    writeln(
      copy(r, 1, 3), '.',
      copy(r, 4, 3), '.',
      copy(r, 7, 3), '-',
      b1, b2
    );
  end;
end.
